Sage/carbon
# fix(form): fix sticky footer width and margin when inside a sidebar

A form inside a sidebar previously hard-coded the margin and padding of the content and footer to match the default padding of a sidebar. This resulted in incorrect styling when custom padding is used for the sidebar as is now allowed. To fix this the padding and margin of the form content and footer now depend on the padding applied to the sidebar.

fix #5404

### Proposed behaviour

The styles (width, padding and margin) applied to form content  and sticky form footer inside a sidebar should depend on the padding applied to the sidebar.



### Current behaviour

The padding and margin of a sticky form footer inside a sidebar are hardcoded and only work properly when the default sidebar padding is used.



### Checklist




#### QA


### Additional context



### Testing instructions



Test form footers - both sticky and not - inside Dialog and DialogFullScreen as well as Sidebar, and when not contained in any modal component, to ensure no visual regressions. Apply custom padding to the parent element if allowed while performing such tests. (Although hopefully any such regressions will be picked up by Chromatic.)

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.
